numpy/ma/core.py

```diff
@@ -2968,8 +2968,8 @@ def __array_wrap__(self, obj, context=None):
                 np.copyto(result, fill_value, where=d)
                 # Update the mask
                 if m is nomask:
-                    if d is not nomask:
-                        m = d
+                    if any(d):
+                        m = True if all(d) else d
```

Comment:
Yeah I've been rethinking these lines. The original point of this PR was to turn an array of False into False.

But for an array of True taken care of here, I probably don't want to turn it into just True. It's possible I want to leave it as an array of True, but `masked` is also a possibility.

I'm also investigating the original problem that this PR is meant to fix (it was making it difficult to reimplement `MaskedArray.var`) to see if there is a better way to fix that altogether.